grails-app/conf/UrlMappings.groovy

```diff
@@ -62,6 +62,8 @@ class UrlMappings {
 
 		"/api/v1/modules"(resources: "moduleApi")
 		"/api/v1/modules/$id/help"(controller: "moduleApi", action: "help")
+		"/api/v1/modules/jsonGetModule"(controller: "moduleApi", action: "jsonGetModule")
```

Comment:
The endpoints should be renamed to align with RESTful principles. Proposal:

- `jsonGetModule` => `POST /api/v1/modules/$id`

Basically this instantiates a module (identified by `id`) with given config (currently passed as parameter `configuration`). In the renamed endpoint, module `id` is provided in the URL, and what is now passed as `configuration` would simply be the request body.

- Add new endpoint: `GET /api/v1/modules/$id`

Would do the same thing as `POST /api/v1/modules/$id` with empty body (return the module with default config).

- `jsonGetModuleTree` => `GET /api/v1/moduleCategories`

Here we have a few options:

1. Don't include the modules in the list of `moduleCategories` (currently included in `jsonGetModuleTree` under key `children`. This would leave it up to the frontend to piece together the module tree by calling both `/api/v1/moduleCategories` and `GET /api/v1/modules`.

2. Always include the modules in the results (as `jsonGetModuleTree` does currently). Here I propose a key change  `children` => `modules`.

3. (preferred) Make the association fetching optional, e.g. `GET /api/v1/moduleCategories?modules=true` would fetch the association, false by default.

Reply:
Are these changes something you want seen in this PR or could these be done later in a separate refactor PR?

Reply:
@harbu they aren't big changes, so I was thinking this PR

Reply:
I did the `jsonGetModule` remapping. W.r.t. to the module categories I went with `GET /api/v1/module_categories` for consistency with rest of API. I believe the `children` may also contain sub-categories not just modules?

Reply:
Swagger also updated.

Reply:
> I believe the children may also contain sub-categories not just modules?

True, good point.